File: python/euler/bin/p0048.py
# project euler: problem 48


def compute(exp: int) -> str:
    modulus = 10**10

    return format(
        sum(pow(x, x, modulus) for x in range(1, exp + 1) if x % 10 != 0) % modulus,
        '010',
    )


def solve() -> str:
    return compute(1_000)

File: typescript/euler/bin/p0048.ts
// project euler: problem 48

export const compute = (exp: bigint): string => {
  const modulus = 10n ** 10n;

  let acc = 0n;
  for (let x = 1n; x <= exp; x++) {
    if (x % 10n !== 0n) {
      acc += (x ** x) % modulus;
    }
  }
  acc = acc % modulus;

  return acc.toString().padStart(10, "0");
};

export const solve = (): string => compute(1_000n);
